controllers/grpcroute_controller.go

```diff
@@ -0,0 +1,376 @@
+	gateway_api_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gateway_api_v1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+	mcs_api "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	"github.com/aws/aws-application-networking-k8s/controllers/eventhandlers"
+	"github.com/aws/aws-application-networking-k8s/pkg/aws"
+	"github.com/aws/aws-application-networking-k8s/pkg/config"
+	"github.com/aws/aws-application-networking-k8s/pkg/deploy"
+	"github.com/aws/aws-application-networking-k8s/pkg/deploy/lattice"
+	"github.com/aws/aws-application-networking-k8s/pkg/gateway"
+	"github.com/aws/aws-application-networking-k8s/pkg/k8s"
+	"github.com/aws/aws-application-networking-k8s/pkg/latticestore"
+	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+	latticemodel "github.com/aws/aws-application-networking-k8s/pkg/model/lattice"
+	lattice_runtime "github.com/aws/aws-application-networking-k8s/pkg/runtime"
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
+// GRPCRouteReconciler reconciles a GRPCRoute object
+type GRPCRouteReconciler struct {
+	client           client.Client
+	scheme           *runtime.Scheme
+	finalizerManager k8s.FinalizerManager
+	eventRecorder    record.EventRecorder
+	modelBuilder     gateway.LatticeServiceBuilder
+	stackDeployer    deploy.StackDeployer
+	latticeDataStore *latticestore.LatticeDataStore
+	stackMashaller   deploy.StackMarshaller
+}
+
+const (
+	grpcRouteFinalizer = "grpcroute.k8s.aws/resources"
+)
+
+func NewGrpcRouteReconciler(
+	cloud aws.Cloud,
+	client client.Client,
+	scheme *runtime.Scheme,
+	eventRecorder record.EventRecorder,
+	finalizerManager k8s.FinalizerManager,
+	latticeDataStore *latticestore.LatticeDataStore,
+) *GRPCRouteReconciler {
+	modelBuilder := gateway.NewLatticeServiceBuilder(client, latticeDataStore, cloud)
+	stackDeployer := deploy.NewLatticeServiceStackDeploy(cloud, client, latticeDataStore)
+	stackMarshaller := deploy.NewDefaultStackMarshaller()
+
+	return &GRPCRouteReconciler{
+		client:           client,
+		scheme:           scheme,
+		finalizerManager: finalizerManager,
+		modelBuilder:     modelBuilder,
+		stackDeployer:    stackDeployer,
+		eventRecorder:    eventRecorder,
+		latticeDataStore: latticeDataStore,
+		stackMashaller:   stackMarshaller,
+	}
+}
+
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=grpcroutes,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=grpcroutes/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=grpcroutes/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the GRPCRoute object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
```

Comment:
It seems quite a bit of code copy pasted, please delete what you dont need

Reply:
Yup!